### DISCUSSION TITLE
Minimal ffmpeg build configure RTSP cameras H264 e HEVC

### DISCUSSION THREAD

**Discussion by eusoubrasileiro** [Motion-Project/motion#1545] [Q&A]
I have been using motion for almost 2 years I guess. I have always built ffmpeg from source due my bad quality IP cameras. 
Now I am trying to make a minimal `ffmpeg` build for motion for SOC (nvidia jetson-nano 2GB)
I only have _h264_ an _hevc_ cameras. And I am using a custom library for hardware accelerate those (`--enable-nvmpi`). That makes me include also decoders `--enable-decoder=hevc_nvmpi --enable-decoder=h264_nvmpi`.

So, I'm trying using the following configure.  
```
./configure --enable-nvmpi --disable-outdevs  --disable-indevs  \
--disable-ffprobe --disable-ffplay --disable-encoders --disable-filters \
--disable-devices --disable-protocols --disable-decoders \
--disable-muxers --disable-demuxers \
--enable-protocol=rtmp \
--enable-encoder=mpeg2video --enable-encoder=mpeg4 \
--enable-muxer=mpeg2video --enable-muxer=mp4 \
--enable-decoder=hevc_nvmpi --enable-decoder=h264_nvmpi \
--enable-muxer=rtsp --enable-muxer=h264 --enable-muxer=hevc --enable-muxer=mp4 \
--enable-demuxer=rtsp --enable-demuxer=h264 --enable-demuxer=hevc  \
--enable-shared --prefix=/usr/local
```
I am also enabling `mpeg2video`, `mpeg4`,  `mp4` encoders/muxers for timelapse recording.
Everything works fine with **ffmpeg** command line like to stream through ffmpeg.  Like, bellow:

```
 ffmpeg -v quiet -stats -rtsp_transport tcp -y -c:v  hevc_nvmpi -i rtsp://user:pass@ipcam-kitchen:554/onvif2 -f null -
```

Unfortunately I'm getting errors with **_timelapse_** recording .  Saying file exists but I check it doesn't exist.  Something like:
```
5271096:Jun 16 22:12:58 jetson-nano motiond[18929]: [2:ml2:garage] [ERR] [ENC] ffmpeg_set_outputfile: error opening file /mnt/motion/movies/timelapse/cam_garage/06_16/2022_06_16_22h_cam_2_garage.mpg: File exists
5271300-Jun 16 22:12:58 jetson-nano motiond[18929]: [2:ml2:garage] [ERR] [ENC] ffmpeg_open: Could not set the stream
5271409-Jun 16 22:12:58 jetson-nano motiond[18929]: [2:ml2:garage] [ERR] [EVT] event_ffmpeg_timelapse: ffopen_open error creating (timelapse) file [/mnt/motion/movies/timelapse/cam_garage/06_16/2022_06_16_22h_cam_2_garage.mpg]
5271628:Jun 16 22:12:58 jetson-nano motiond[18929]: [5:ml5:new] [ERR] [ENC] ffmpeg_set_outputfile: error opening file /mnt/motion/movies/timelapse/cam_new/06_16/2022_06_16_22h_cam_5_new.mpg: File exists
5271823-Jun 16 22:12:58 jetson-nano motiond[18929]: [5:ml5:new] [ERR] [ENC] ffmpeg_open: Could not set the stream
5271929-Jun 16 22:12:58 jetson-nano motiond[18929]: [5:ml5:new] [ERR] [EVT] event_ffmpeg_timelapse: ffopen_open error creating (timelapse) file [/mnt/motion/movies/timelapse/cam_new/06_16/2022_06_16_22h_cam_5_new.mpg]
```

### Someone knows what else is required I am missing on `configure` to make timelapse work in this minimal ffmpeg build?

**Comment by tosiara** [comment]
Never seen that and never tried compiling ffmpeg that way
The only I could suggest is to try compiling ffmpeg in a default way and see if that ever works. Then, try adding your custom flags one by one until you find which one breaks the things

**Comment by eusoubrasileiro** [comment]
@tosiara  thanks you are always willing to help us! Yeah as it is working the default way without problems. It's just a 'branch on my repository' trying to make ffmpeg build faster and optimized.
I will keep on trying that way then. When I finish I will post here the final `configure` options.

**Comment by eusoubrasileiro** [comment]
Updating with what works for far.
Note that I only have h264 an hevc RTSP cameras and I use `nvpmi` library for hardware accelerated decoding. 

```
./configure --enable-nvmpi \
--disable-ffprobe --disable-ffplay \
--disable-outdevs --disable-indevs --disable-devices --disable-filters \
--enable-shared --prefix=/usr/local \
--disable-encoders \
--enable-encoder=mpeg2video --enable-encoder=mpeg4 \
--disable-muxers \
--enable-muxer=mpeg2video --enable-muxer=mp4 --enable-muxer=rtsp \
--enable-muxer=h264 --enable-muxer=hevc --enable-muxer=mp4 \
--disable-decoders \
--enable-decoder=hevc_nvmpi --enable-decoder=h264_nvmpi \
--enable-decoder=hevc --enable-decoder=h264
```
